lib/super_digit.rb

```diff
@@ -1,15 +1,31 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity O(n)
+# Space Complexity Constant
 def super_digit(n)
-   
+  return n if n < 10
+  s_digit = 0
+  while n >= 10
+    s_digit += n % 10
+    n /= 10
+  end
+
+  return super_digit(s_digit)
 end
   
 
 # Time Complexity - ?
 # Space Complexity - ?
 def refined_super_digit(n, k)
```

Comment:
This works, but it will take O((nk)^2).  You can do better with recursive programming O(d^2 + k^2)